src/main/resources/data-repository-openapi.yaml

```diff
@@ -446,10 +550,58 @@ definitions:
         items:
           $ref: '#/definitions/DatasetSourceModel'
 
+  DatasetRequest:
+    description: |
+      complete dataset save the dataset id, for dataset creation
+      Schema is the shape of the resulting dataset: the table names and columns in the
+      tables. The datatypes are derived from the underlying data.
+      DatasetData provides the rows from the asset that are to be included in the
+      dataset. DatasetMapping instructs how to map columns from tables on the columns
+      in the dataset schema.
+      data_handling: immutable means we get explicit list of rows to show
+      passthru means we don't filter rows and take whatever is there
+    type: object
+    required:
+      - name
+      - schema
+      - data
+      - source
+    properties:
+      name:
+        $ref: '#/definitions/DatasetNameModel'
+      description:
+        type: string
+        description: Description of the dataset
+      data_handling:
```

Comment:
i don't think this is part of the MVP. at least i don't understand the requirement for this.